tests/bwc/test_upgrade.py

```diff
@@ -303,15 +305,26 @@ def assert_data_persistence(self,
                 cursor.execute(f'select * from versioned."{table}"')
                 cursor.execute(f'insert into versioned."{table}" (id, col_int) values (?, ?)', [str(uuid4()), 1])
 
+            # to trigger `alter` stmt bug(https://github.com/crate/crate/pull/17178) that falsely updated the table's
+            # version created setting that resulted in oids replacing column names in resultsets
+            cursor.execute('ALTER TABLE doc.parted SET ("refresh_interval" = 900)')
+
             # older versions had a bug that caused this to fail
-            if version in ('latest-nightly', '3.2'):
-                # Test that partition and dynamic columns can be created
-                obj = {"t_" + version.replace('.', '_'): True}
-                args = (str(uuid4()), version, obj)
-                cursor.execute(
-                    'INSERT INTO doc.parted (id, version, cols) values (?, ?, ?)',
-                    args
-                )
+            # Test that partition and dynamic columns can be created
+            key = "t_" + version.replace('.', '_')
+            obj = {key: True}
+            args = (str(uuid4()), version, obj)
+            cursor.execute(
+                'INSERT INTO doc.parted (id, version, cols) values (?, ?, ?)',
+                args
+            )
+            accumulated_dynamic_column_names.append(key)
+            cursor.execute('SELECT cols from doc.parted')
+            result = cursor.fetchall()
+            for row in result:
+                if row[0] is not None:
```

Comment:
This shouldn't happen, right? wouldn't be better to remove the check and have it throw an error if the result is `null`?

Also, please do a `REFRESH` after the INSERT, to make sure that we test all the values for `cols`.